### PR TITLE
[DOC] Update CONTRIBUTING and README with developer installation, contributing and testing instructions

### LINKED ISSUE #346: Update CONTRIBUTING with detailed process for dev setup, branching, submitting PRs
### Summary
New developers contributing to tedana should be able to follow an intuitive process for submitting a PR. To achieve this, CONTRIBUTING could contain a detailed process suggesting the steps a developer could follow, from the developer setup to opening an issue to submitting a PR, according to `tedana` best practices.

### Additional Detail
Currently in CONTRIBUTING, the [Making a change](https://github.com/ME-ICA/tedana/blob/master/CONTRIBUTING.md#making-a-change) section contains several steps instructing a developer how to go about creating issues, making changes and submitting a pull request. However, the developer setup required to do this is not immediately clear. Also, the preferred process for forking, branching, merging, etc, could be more intuitive for newcomers, e.g. a visual diagram or step-by-step explanation with accompanying git commands could help lower the barrier to entry. 

### Next Steps
- [ ] Tedana developers to share input on the required steps (Suggestion from @jbteves: fork -> clone fork -> branch from master -> open PR from branch -> merge PR -> update master -> delete branch)
- [ ] Create PR with suggested changes to CONTRIBUTING @jsheunis

### PR DESCRIPTION
**Pull request by jsheunis**
Closes #346 

Changes proposed in this pull request:

README:
- Installation instructions were given more structure with headings. Divided into installation for:
   - local python environment,
   - conda environment
   - developer setup (which now links to updated CONTRUBTING, but could in future link to fully separate DEVELOPMENT_GUIDE à la [scona](https://github.com/WhitakerLab/scona/blob/master/DEVELOPMENT_GUIDE.md#building-sphinx-docs))

CONTRIBUTING:
- Improved heading structure for the [Making a change](https://github.com/ME-ICA/tedana/blob/master/CONTRIBUTING.md#making-a-change) section
- Instructions on developer setup
- Included instructions for testing docs. Could link in future to the worked testing/contributing example #372

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -116,39 +116,57 @@ The current list of labels are [here][link_labels] and include:
...
-Try to keep the changes focused. We've found that working on a [new branch][link_branches] makes it easier to keep your changes targeted.
+### 3. Run the developer setup
 
-When you're creating your pull request, please make sure to review the tedana [style conventions](#style-guide).
-
-Note that to test a change, you may need to set up your local repository to run a tedana workflow. 
+To test a change, you may need to set up your local repository to run a `tedana` workflow. 
 To do so, run
 ```
 python setup.py develop
```
Comment (jbteves) [inline CONTRIBUTING.md]:
```suggestion
pip install -e
```

Comment (KirstieJane) [reply CONTRIBUTING.md]: Can you add the full command @jbteves? Probably the GitHub link...

Comment (jbteves) [reply CONTRIBUTING.md]: Added the dot since the next line says "from your repository."

Comment (tsalo) [reply CONTRIBUTING.md]: Why `pip install -e .` over `python setup.py develop`? I normally use `setup.py` myself.

Comment (tsalo) [reply CONTRIBUTING.md]: I just saw @jsheunis' comment in the corresponding issue. I assume that's why. Never mind!

**Comment by emdupre** [comment]
This is great, thanks @jsheunis ! :sunflower: 

Just wanted to ping since it looks like there's only one small suggested revision, and I want to make sure this doesn't go stale. Let us know if we can help in any way !

**Comment by emdupre** [comment]
@all-contributors please add @jsheunis for documentation !

**Comment by jsheunis** [comment]
I'm a bit new to the process of reviews, suggestions, revisions, etc. I committed the change that @jbteves suggested, hope that was the right thing to do and the right way to do it!

**Comment by jbteves** [comment]
@jsheunis that's fine. When a reviewer makes a code suggestion GitHub offers the option to automatically create a commit and add it to the branch with the reviewer as co-author. If you make and push an otherwise identical commit manually on your own machine then the reviewer won't be co-authored unless you add a tag. That's the only real difference. I seriously doubt anybody will get bent out of shape about it. The nice thing with a suggested change is that it's just one click to add it if the original contributor agrees.

**Comment by emdupre** [comment]
This is great, thanks @jsheunis ! Committing the suggested change is perfect in my book.

@jbteves can you update your review if you're ok with it as-is and we'll get this merged ?

**Pull request merged**